internal/controller/agentpool_controller_autoscaling.go

```diff
@@ -61,9 +65,10 @@ func matchWildcardName(wildcard string, str string) bool {
 	}
 }
 
-// pendingWorkspaceRuns returns the number of workspaces with pending runs for a given agent pool.
+// pendingWorkspaceRuns returns the number of agents needed to execute current pending runs for a given agent pool.
+// If there are no plan-only runs in the list of current pending runs for a workspace this functoion returns the number of workspaces.
```

Comment:
Shall we stick to the number of pending runs only in the description? I would suggest either removing the second line or rephrasing it, as it’s confusing in its current form.